### ISSUE TITLE
YAML parser: accessibility check failure

### ISSUE THREAD

**Issue by ethindp** [stcarrez/dynamo#17]
The YAML parser contains this line in yaml-parser.ads:

```ada
package Tag_Handle_Sets is new Yaml.Text_Set (Text.Reference);
```

I'm not positive, but GNAT is noting that an accessibility check is failing and that `Program_Error` will be raised at runtime (presumably during elaboration). I'm not sure if this is the actual line that's triggering the check, I can keep digging to try and find out where the real error is.

**Comment by stcarrez** [comment]
Yes, but the exception is in fact not raised, I guess because the faulty operation does not get called.

The parser is based on https://github.com/yaml/AdaYaml

The original AdaYaml sources are buggy: the original implementation tried to do weird things with the memory. It does not work and crashed.  I've replaced that memory manager with the standard `Unbounded_String`.

What now remains is this annoying warning....

**Comment by stcarrez** [comment]
If we compile with FSF Debian GNAT 9.4.0 the warning does not occur.

**Comment by stcarrez** [comment]
Thanks for the bug report. I've managed to fix it by avoiding returning the access type in the `Get` function.
It was some kind of a hack that the `Get` function returned an access so that the associated value could be modified. I've introduced a `Update` and used it instead.

**Comment by ethindp** [comment]
Sorry I didn't find that myself, college has been consuming all of my time (ugh). I've got some free time now though at least.

**Comment by stcarrez** [comment]
It's ok! It's not easy to digg into this yaml parser because there was some use of memory pools that was buggy. And, well, it takes time for a newcomer (or even me) to enter in this...
